analysis/filter/StructureModifier/src/test/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/NodeRemoverTest.kt

```diff
@@ -0,0 +1,103 @@
+
+import de.maibornwolff.codecharta.model.Project
+import de.maibornwolff.codecharta.serialization.ProjectDeserializer
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.io.File
+
+class NodeRemoverTest {
+    private lateinit var sampleProject: Project
+
+    @BeforeEach
+    fun serializeProject() {
+        val bufferedReader = File("src/test/resources/sample_project.cc.json").bufferedReader()
+        sampleProject = ProjectDeserializer.deserializeProject(bufferedReader)
+    }
+
+    @Test
+    fun `Non existent path leads to replication of the project`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/somethig"))
+
+        Assertions.assertThat(result).isEqualToComparingFieldByFieldRecursively(sampleProject)
+    }
+
+    @Test
+    fun `Non affected nodes are kept`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/src/main"))
+
+        val testFolder = result.rootNode.children.first().children.first()
+        Assertions.assertThat(testFolder.name).isEqualTo("test")
+    }
+
+    @Test
+    fun `specified nodes are removed`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/src/main"))
+
+        val srcContent = result.rootNode.children.first().children
+        Assertions.assertThat(srcContent.size).isEqualTo(2)
+        Assertions.assertThat(srcContent.map { it.name }).doesNotContain("main")
+    }
+
+    @Test
+    fun `multiple nodes are removed`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/src/main/file1.java", "root/src/folder3/"))
+
+        val srcContent = result.rootNode.children.first().children
+        val mainContent = srcContent.first().children
+        Assertions.assertThat(srcContent.map { it.name }).doesNotContain("folder3")
+        Assertions.assertThat(mainContent.map { it.name }).containsOnly("file2.java")
+    }
+
+    @Test
+    fun `project name is set to specified name`() {
+        val subProjectExtractor = NodeRemover(sampleProject, "fooBar")
+
+        val result = subProjectExtractor.remove(arrayOf())
+
+        Assertions.assertThat(result.projectName).isEqualTo("fooBar")
+    }
+
+    @Test
+    fun `Affected edges are removed`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/foo"))
+
+        Assertions.assertThat(result.edges.size).isEqualTo(1)
+        Assertions.assertThat(result.edges.first()).isEqualToComparingFieldByFieldRecursively(sampleProject.edges.last())
+        Assertions.assertThat(result.edges.map { it.fromNodeName }).doesNotContain("/roo/foo")
```

Comment:
Typo: "/roo/foo" -> "/root/foo". See next line as well

Reply:
good catch, test does not make a lot of sense otherwise :)